Building cquery with alternative c standard libraries

When trying to compile [cquery](https://github.com/cquery-project/cquery) on Void Linux with [musl libc](https://www.musl-libc.org/) I get the following error.

```
loguru.hpp:1391:11: fatal error: execinfo.h: No such file or directory
  #include <execinfo.h>  // for backtrace
```

`execinfo.h` is `glibc` specific and prevents compilation when using alternative standard libraries. I have this [pending issue](https://github.com/emilk/loguru/issues/69) for loguru asking for support but it hasn't gotten much attention so far. 

Does cquery strictly depend on loguru or is it possible to build without it? Is it possible to substitute it for another logging library? It's a shame that a single, arguably minor dependency prevents cquery from being used with any c standard library besides `glibc`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -131,8 +131,14 @@ elseif(${CMAKE_SYSTEM_NAME} STREQUAL Windows)
 endif()
 
 ### Definitions
+file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/foo.cpp
+     "#include <cxxabi.h>\n#include <dlfcn.h>\n#include <execinfo.h>\nint main(){}")
+try_compile(LOGURU_STACKTRACES_ALLOWED ${CMAKE_CURRENT_BINARY_DIR}
+            SOURCES ${CMAKE_CURRENT_BINARY_DIR}/foo.cpp)
+file(REMOVE ${CMAKE_CURRENT_BINARY_DIR}/foo.cpp)
 
 target_compile_definitions(cquery PRIVATE
+  LOGURU_STACKTRACES=$<BOOL:${LOGURU_STACKTRACES_ALLOWED}>
   LOGURU_WITH_STREAMS=1
   LOGURU_FILENAME_WIDTH=18
   LOGURU_THREADNAME_WIDTH=13

